### DISCUSSION TITLE
React Devtools and webpack config

### DISCUSSION THREAD

**Discussion by Dev-rdg** [vercel/next.js#39861] [Help]
Hello.

I was having trouble debugging a component and turned to react devtools. 
My problem is that with the default webpack config the parsing of hook names fails.
I know it's not recommended to alter devtool options for performance reasons, and i know next revert changes made when starting the dev server.
Nonetheless when i tried setting the devtool option to "cheap-module-source-map" and commenting the part of the code responsible for reverting it; the parsing of hook did start to work again.
Is there any other way to achieve what i want or am I on my own?

**Comment by icyJoseph** [comment]
Mmm perhaps you could use https://reactjs.org/docs/hooks-reference.html#usedebugvalue? This way you can inspect with mode detail, and specifically only those hooks that you need to debug. 

Otherwise, I think you'd need to serve the source maps, which would be parsed by the devTools, and activate the `Always parse hook names from source (may be slow)` option in the React DevTools. You might also want to take a look at, https://nextjs.org/docs/advanced-features/source-maps.

What are you trying to debug, and how exactly?
